### PR TITLE
[SuperNode program] task: update supernode url.

### PR DESCRIPTION
**Pull request by AnthonyLaw**
## What was the issue?
- supernode URL is no longer valid, it needs to replace the new URL.

## What's the fix?
- removed `getGeolocation` because it's an unused method.
- add `getNodes` method in superNodeProgram service.
- replace supernode's `getNodes` in normal and multisig's ImportanceTransfer controller.
- add unit test.

### CODE REVIEW

File: src/app/modules/importanceTransfer/multisig/importanceTransfer.controller.js [hunk 1]
```diff
@@ -98,10 +99,16 @@ class MultisigImportanceTransferCtrl {
             this.nodes = this._Nodes.get();
         } else {
             this.nodes = [];
-            // Get supernodes
-            nem.com.requests.supernodes.all().then((data) => {
+            // Get superNodes with pageNumber 0
```
Comment (cryptoBeliever) [inline src/app/modules/importanceTransfer/multisig/importanceTransfer.controller.js]: Why just the first page? We can take all of them (as the SN page is doing - take elements page by page until there is no more) and get random X elements. I would say X = 50 for example (I will have a look if it looks good with that amount of elements).

File: src/app/modules/importanceTransfer/multisig/importanceTransfer.controller.js [hunk 2]
```diff
@@ -98,10 +99,16 @@ class MultisigImportanceTransferCtrl {
...
+            this._superNodeProgram.getNodes(0).then((data) => {
                 this._$timeout(() => {
-                    this.nodes = this._$filter('toEndpoint')(data.nodes);
+                    // shuffle nodes
+                    data.sort(() => Math.random() - 0.5);
+
+                    this.nodes = data.map((node) => {
+                        const { protocol, hostname, port } = new URL(node.endpoint);
+                        return nem.model.objects.create("endpoint")(`${protocol}//${hostname}`, Number(port))
+                    })
```
Comment (Jaguar0625) [inline src/app/modules/importanceTransfer/multisig/importanceTransfer.controller.js]: please move this logic to superNodeProgram so it is in single place. you can call it something like getRandomNode(s) and call it from both places

File: tests/specs/superNodeProgram.spec.js [hunk 3]
```diff
@@ -272,4 +272,92 @@ describe('SuperNodeProgram service tests', () => {
             done();
         });
     });
+
+    describe('getNodes()', () => {
+        it('should fetch to be called with valid endpoint url when pageNumber=0 passed', async (done) => {
```
Comment (Jaguar0625) [inline tests/specs/superNodeProgram.spec.js]: should fetch to be called => should call fetch

File: tests/specs/MultisigImportanceTransferModule.spec.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+                },
+                {
+                    "host": "http://localhost",
+                    "port": 7890
+                }
+            ]);
+            done();
+        });
+    });
+});
```
Comment (Jaguar0625) [inline tests/specs/MultisigImportanceTransferModule.spec.js]: newline at eof

File: tests/specs/MultisigImportanceTransferModule.spec.js [hunk 5]
```diff
@@ -0,0 +1,188 @@
+import WalletFixture from '../data/wallet';
```
Comment (Jaguar0625) [inline tests/specs/MultisigImportanceTransferModule.spec.js]: should filename be lowerCamelCase like others ?

File: tests/specs/MultisigImportanceTransferModule.spec.js [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        DataBridge.accountData = AccountDataFixture.testnetAccountData;
+    }
+
+    function createDummyWalletContextMainnet(Wallet) {
+        Wallet.use(WalletFixture.mainnetWallet);
+        Nodes.setDefault();
+        DataBridge.accountData = AccountDataFixture.mainnetAccountData;
+    }
+
+    describe('setNodes', () => {
```
Comment (Jaguar0625) [inline tests/specs/MultisigImportanceTransferModule.spec.js]: please refactor as these tests are identical to those in previous file - this is generally an indication of a code smell / refactoring opportunity

File: src/app/modules/importanceTransfer/multisig/importanceTransfer.controller.js (hunk 1 above)
Comment (AnthonyLaw) [reply src/app/modules/importanceTransfer/multisig/importanceTransfer.controller.js]: I forgot the apply recursively fetch on this. I will refactor this.

File: tests/specs/MultisigImportanceTransferModule.spec.js (hunk 6 above)
Comment (AnthonyLaw) [reply tests/specs/MultisigImportanceTransferModule.spec.js]: yeah, I will work on this refactor.

**Review by cryptoBeliever — APPROVED** [review]

**Comment by cryptoBeliever** [comment]
@AnthonyLaw I don't see any issue with wallet for both multisig and single harvesting view :+1:

**Review by Jaguar0625 — APPROVED** [review]
nice job 👍

**Review by cryptoBeliever — APPROVED** [review]

**Pull request merged**